VirtoCommerce.Storefront/Domain/Security/OrderStatusChangeHandler.cs

```diff
@@ -0,0 +1,33 @@
+namespace VirtoCommerce.Storefront.Domain.Security
+{
+    using System.Linq;
+    using System.Threading.Tasks;
+
+    using Microsoft.AspNetCore.Authorization;
+
+    using VirtoCommerce.Storefront.AutoRestClients.OrdersModuleApi.Models;
+    using VirtoCommerce.Storefront.Model;
+    using VirtoCommerce.Storefront.Model.Security;
+
+    public class OrderStatusChangeHandler : AuthorizationHandler<OrderStatusChangeAuthorizationRequirement, CustomerOrder>
```

Comment:
I  suppose that we need a one common `AuthorizationHandler`  for  check  authorization based on the permissions policies.  That can be used as method attribute `[Authorization('permission-name')` 

See how this has done in the platform v3:

https://github.com/VirtoCommerce/vc-platform-core/blob/dev/VirtoCommerce.Platform.Web/Controllers/Api/ModulesController.cs#L58
https://github.com/VirtoCommerce/vc-platform-core/blob/dev/VirtoCommerce.Platform.Security/Authorization/PermissionAuthorizationPolicyProvider.cs
https://github.com/VirtoCommerce/vc-platform-core/blob/dev/VirtoCommerce.Platform.Security/Authorization/PermissionAuthorizationHandlerBase.cs